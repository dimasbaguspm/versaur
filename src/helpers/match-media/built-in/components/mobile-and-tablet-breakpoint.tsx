import { BREAKPOINT_MOBILE_AND_TABLET } from '../../constants'
import { MatchMedia } from '../../match-media'

/**
 * Props for breakpoint components
 */
interface BreakpointComponentProps {
  children: React.ReactNode
}

/**
 * Component that only renders children on mobile and tablet viewports
 *
 * @example
 * ```tsx
 * <MobileAndTabletBreakpoint>
 *   <MobileAndTabletContent />
 * </MobileAndTabletBreakpoint>
 * ```
 */
export function MobileAndTabletBreakpoint({
  children,
}: BreakpointComponentProps) {
  return (
    <MatchMedia query={BREAKPOINT_MOBILE_AND_TABLET}>{children}</MatchMedia>
  )
}
