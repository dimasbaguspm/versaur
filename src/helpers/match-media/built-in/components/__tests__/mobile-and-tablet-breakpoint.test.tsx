import { render, screen } from '@testing-library/react'
import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest'
import { MobileAndTabletBreakpoint } from '../mobile-and-tablet-breakpoint'
import {
  createMatchMediaMock,
  setupMatchMediaMock,
} from '../../../test-helpers'

describe('MobileAndTabletBreakpoint', () => {
  const { beforeEach: setupBefore, afterEach: setupAfter } =
    setupMatchMediaMock()

  beforeEach(() => {
    setupBefore()
  })

  afterEach(() => {
    setupAfter()
  })

  it('should render children when viewport is mobile or tablet', () => {
    const mockMatchMedia = createMatchMediaMock(true)
    window.matchMedia = vi.fn(() => mockMatchMedia as unknown as MediaQueryList)

    render(
      <MobileAndTabletBreakpoint>
        <div data-testid='mobile-tablet-content'>Mobile/Tablet Content</div>
      </MobileAndTabletBreakpoint>
    )

    expect(screen.getByTestId('mobile-tablet-content')).toBeInTheDocument()
  })

  it('should not render children when viewport is desktop', () => {
    const mockMatchMedia = createMatchMediaMock(false)
    window.matchMedia = vi.fn(() => mockMatchMedia as unknown as MediaQueryList)

    render(
      <MobileAndTabletBreakpoint>
        <div data-testid='mobile-tablet-content'>Mobile/Tablet Content</div>
      </MobileAndTabletBreakpoint>
    )

    expect(
      screen.queryByTestId('mobile-tablet-content')
    ).not.toBeInTheDocument()
  })

  it('should match snapshot when rendered', () => {
    const mockMatchMedia = createMatchMediaMock(true)
    window.matchMedia = vi.fn(() => mockMatchMedia as unknown as MediaQueryList)

    const { asFragment } = render(
      <MobileAndTabletBreakpoint>
        <div>Mobile/Tablet Content</div>
      </MobileAndTabletBreakpoint>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
