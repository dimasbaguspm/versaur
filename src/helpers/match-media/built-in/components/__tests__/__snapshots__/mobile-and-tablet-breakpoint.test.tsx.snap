// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MobileAndTabletBreakpoint > should match snapshot when rendered 1`] = `
<DocumentFragment>
  <div>
    Mobile/Tablet Content
  </div>
</DocumentFragment>
`;
